src/generic-container/generic-container.test.ts

```diff
@@ -509,6 +510,39 @@ describe("GenericContainer", () => {
     await container.stop();
   });
 
+  it("should create container with volume", async () => {
+    const volume: Volume = { name: "/volumes/new-volume" };
+
+    const container = await new GenericContainer("cristianrgreco/testcontainer:1.1.12")
+      .withExposedPorts(8080)
+      .withVolumes(volume)
+      .start();
+
+    const dockerContainer = await getContainerById(container.getId());
+    const containerInfo = await dockerContainer.inspect();
+    await container.stop();
+    expect(containerInfo.Mounts.length).toBe(1);
```

Comment:
Could this be tested similar to the bind mounts, where the actual behaviour is tested, instead of checking the container info? I.e you should be able to mount a volume and interact with it and assert its values

Reply:
Good idea, on it.